python/cudf/cudf/core/column/column.py

```diff
@@ -1301,11 +1365,12 @@ def column_empty_like(
     ):
         column = cast("cudf.core.column.CategoricalColumn", column)
         codes = column_empty_like(column.codes, masked=masked, newsize=newsize)
+
         return build_column(
             data=None,
             dtype=dtype,
             mask=codes.base_mask,
-            children=(as_column(codes.base_data, dtype=codes.dtype),),
+            children=(codes,),
```

Comment:
Was the old version just unnecessary? Seems like no reason for the `as_column` call.

Reply:
Yeah, it was redundant.